cmd/dmsgpost/dmsgpost.go

```diff
@@ -0,0 +1,226 @@
+	rootCmd.Flags().StringVarP(&dmsgpostAgent, "agent", "a", "dmsgpost/"+buildinfo.Version(), "identify as `AGENT`")
+	if os.Getenv("dmsgpost_SK") != "" {
+		sk.Set(os.Getenv("dmsgpost_SK")) //nolint
+	}
+	rootCmd.Flags().VarP(&sk, "sk", "s", "a random key is generated if unspecified\n\r")
+	var helpflag bool
+	rootCmd.SetUsageTemplate(help)
+	rootCmd.PersistentFlags().BoolVarP(&helpflag, "help", "h", false, "help for "+rootCmd.Use)
+	rootCmd.SetHelpCommand(&cobra.Command{Hidden: true})
+	rootCmd.PersistentFlags().MarkHidden("help") //nolint
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "dmsgpost",
+	Short: "dmsgpost",
+	Long: `
+	┌┬┐┌┬┐┌─┐┌─┐┌─┐┌─┐┌─┐┌┬┐
+ 	 │││││└─┐│ ┬├─┘│ │└─┐ │
+	─┴┘┴ ┴└─┘└─┘┴  └─┘└─┘ ┴ `,
+	SilenceErrors:         true,
+	SilenceUsage:          true,
+	DisableSuggestions:    true,
+	DisableFlagsInUseLine: true,
+	Version:               buildinfo.Version(),
+	PreRun: func(cmd *cobra.Command, args []string) {
+		if dmsgDisc == "" {
+			dmsgDisc = skyenv.DmsgDiscAddr
+		}
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		if dmsgpostLog == nil {
+			dmsgpostLog = logging.MustGetLogger("dmsgpost")
+		}
+		if logLvl != "" {
+			if lvl, err := logging.LevelFromString(logLvl); err == nil {
+				logging.SetLevel(lvl)
+			}
+		}
+
+		ctx, cancel := cmdutil.SignalContext(context.Background(), dmsgpostLog)
+		defer cancel()
+
+		pk, err := sk.PubKey()
+		if err != nil {
+			pk, sk = cipher.GenerateKeyPair()
+		}
+
+		u, err := parseURL(args)
+		if err != nil {
+			dmsgpostLog.WithError(err).Fatal("failed to parse provided URL")
+		}
+
+		dmsgC, closeDmsg, err := startDmsg(ctx, pk, sk)
+		if err != nil {
+			dmsgpostLog.WithError(err).Fatal("failed to start dmsg")
+		}
+		defer closeDmsg()
+
+		httpC := http.Client{Transport: dmsghttp.MakeHTTPTransport(ctx, dmsgC)}
+
+		req, err := http.NewRequest(http.MethodPost, u.URL.String(), strings.NewReader(dmsgpostData))
+		if err != nil {
+			dmsgpostLog.WithError(err).Fatal("Failed to formulate HTTP request.")
+		}
+		req.Header.Set("Content-Type", "text/plain")
+
+		resp, err := httpC.Do(req)
+		if err != nil {
+			dmsgpostLog.WithError(err).Fatal("Failed to execute HTTP request.")
+		}
+
+		defer func() {
+			if err := resp.Body.Close(); err != nil {
+				dmsgpostLog.WithError(err).Fatal("Failed to close response body")
+			}
+		}()
+		respBody, err := io.ReadAll(resp.Body)
+		if err != nil {
+			dmsgpostLog.WithError(err).Fatal("Failed to read respose body.")
+		}
+		fmt.Println(string(respBody))
+	},
+}
+
+// URL represents a dmsg http URL.
+type URL struct {
+	dmsg.Addr
+	url.URL
+}
+
+// Fill fills the internal fields from an URL string.
+func (du *URL) fill(str string) error {
+	u, err := url.Parse(str)
+	if err != nil {
+		return err
+	}
+
+	if u.Scheme == "" {
+		return errors.New("URL is missing a scheme")
+	}
+
+	if u.Host == "" {
+		return errors.New("URL is missing a host")
+	}
+
+	du.URL = *u
+	return du.Addr.Set(u.Host)
+}
```

Comment:
Can we follow the example from stdlib `Parse()`and return the parsed url instead of calling a method on it? I dont really see why we do it this way...

Reply:
it's just copied from dmsgget

https://github.com/skycoin/dmsg/blob/develop/cmd/dmsgget/dmsgget.go#L180